sssd_test_framework/roles/ad.py

```diff
@@ -2331,3 +2395,387 @@ def print(self) -> str:
 
 
 ADNetgroupMember: TypeAlias = LDAPNetgroupMember[ADUser, ADNetgroup]
+
+
+class ADCertificateAuthority(GenericCertificateAuthority):
+    """
+    Certificate Authority server management.
+
+    This class allows requesting, revoking, placing/removing certificate holds,
+    and retrieving certificate information via certreq and certutil commands.
+    """
+
+    def __init__(self, host: ADHost) -> None:
+        """
+        Initialize the AD Certificate Authority helper.
+
+        :param host: Remote AD host.
+        :type host: ADHost
+        """
+        self.host = host
+        self.cli = host.cli
+        self.temp_dir = f"C:\\pki\\ad_test_certs_{os.getpid()}_{uuid.uuid4().hex}"
+        self._create_temp_dir()
+
+    def _create_temp_dir(self) -> None:
+        """Create temporary directory for certificate files."""
+        self.host.conn.run(f'New-Item -ItemType Directory -Path "{self.temp_dir}" -Force')
+
+    def __request_enrollment(
+        self,
+        template: str,
+        subject: str,
+        key_size: int = 2048,
+    ) -> tuple[str, str, str]:
+        """
+        Request a certificate enrollment (enrollment certificate).
+
+        This method prepares an INF file and requests an enrollment certificate.
+        It is intended for internal use, e.g., by smartcard certificate requests.
+
+        :param template: Certificate template name (e.g., "User").
+        :type template: str
+        :param subject: Certificate subject (e.g., "CN=testuser").
+        :type subject: str
+        :param key_size: RSA key size in bits.
+        :type key_size: int
+        :returns: A tuple of (certificate_path, key_path, csr_path).
+        :rtype: tuple[str, str, str]
+        :raises RuntimeError: If certificate request fails.
```

Comment:
This is the first time I'm seeing the RunTimeError docstring, I think it's a great idea, thanks.